src/iarc7_abstract/roomba_request_executer.py

```diff
@@ -159,38 +231,61 @@ def _run(cls, roomba_request, status_callback):
                 cls._client.cancel_goal()
                 rospy.logwarn("Hold Position Task canceled")
 
-            # state transitioning
-            if (state != RoombaRequestExecuterState.HOLD_POSITION
-                    and (not cls._client.get_result()
-                        or not cls._client.get_result().success)):
-                if state == RoombaRequestExecuterState.RECOVER_FROM_FAILURE:
-                    state = RoombaRequestExecuterState.FAILED_TASK_AND_RECOVERY
-                elif state == RoombaRequestExecuterState.RECOVER_FROM_SUCCESS:
-                    state = RoombaRequestExecuterState.FAILED_RECOVERY
-                else:
-                    state = RoombaRequestExecuterState.RECOVER_FROM_FAILURE
-            else:
+            """state transitioning"""
+
+            # see http://docs.ros.org/api/actionlib_msgs/html/msg/GoalStatus.html
+            # for status IDs
+            if cls._client.get_state() not in [4,5,9]:
```

Comment:
Can you replace the numbers here with variables that say what these are, or just say what those states are in the comment?

Reply:
added comment